snapcraft.yaml

```diff
@@ -2,27 +2,41 @@ name: pi
 version: 20-1
 summary: Raspberry Pi gadget
 description: |
- Support files for booting Raspberry Pi.
- This gadget snap supports the Raspberry Pi 2B, 3B, 3A+, 3B+, 4B, Compute
- Module 3, and the Compute Module 3+ universally.
+  Support files for booting Raspberry Pi.
+  This gadget snap supports the Raspberry Pi 2B, 3B, 3A+, 3B+, 4B, Compute
+  Module 3, and the Compute Module 3+ universally.
 type: gadget
 base: core20
 architectures:
   - build-on: [amd64, armhf]
     run-on: armhf
+
+package-repositories:
+  - type: apt
+    ppa: snappy-dev/image
+
 confinement: strict
 grade: stable
 parts:
   gadget:
     plugin: nil
+    source: .
     override-build: |
+      set -x
+      if [ "$SNAPCRAFT_ARCH_TRIPLET" = "x86_64-linux-gnu" ] || [ -z "$SNAPCRAFT_ARCH_TRIPLET" ]; then
+        # manually set since new snapcraft doesn't like --target-arch when 
+        # building with lxd/multipass
+        BUILD_ARCH_TRIPLET=arm-linux-gnueabihf
+      else
+        BUILD_ARCH_TRIPLET=$SNAPCRAFT_ARCH_TRIPLET
+      fi
       if [ "$(arch)" != "aarch64" -a "$(arch)" != "armv7l" ]; then
         echo "Cross compilation detected; using pre-defined sources list"
-        OPTIONAL_ARGS="SOURCES_HOST=\"./helpers/sources.list.cross\""
+        OPTIONAL_ARGS="SOURCES_HOST=\"./helpers/cross-sources.list\" SOURCES_D_HOST=\"./helpers\""
```

Comment:
The sources.list is arch qualified, and is correct for either native or cross compilation, does it need to be in a conditional?

Reply:
yes you're right it can always be used